desktop/libs/indexer/src/indexer/api3_tests.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import sys
+from nose.tools import assert_equal
+from django.utils.datastructures import MultiValueDict
+from django.core.files.uploadhandler import InMemoryUploadedFile
+
+from desktop.settings import BASE_DIR
+from indexer.api3 import upload_local_file
+from urllib.parse import urlparse, unquote as urllib_unquote
```

Comment:
This import might fail for py2 and we might need to add this in the `sys.version_info[0]` if check (adding for py2 equivalent also).

Reply:
done